### DISCUSSION TITLE
FR: Provide bulk actions (Install/Disable) for plugins

### DISCUSSION THREAD

**Discussion by engram-design** [craftcms/cms#8498] [Ideas]
Often we have 10-15 plugins that are installed on each site. Currently (with Craft 2) we have to quickly click the "Install" button on each plugin before the page reloads. However, with Craft 3 changing this Install button into a dropdown, this isn't the best feasible. Its also generally not the best approach in the first place!

I realise the code to actually install all plugins is trivial, but it'd be great to have a way to handle these bulk options. This may be useful for debugging (disable all plugins). Potentially, in a similar way to the Entries index, where you can check multiple plugins and have an element actions-like dropdown to perform bulk actions on plugins (Install, Uninstall, Enable, Disable).

**Comment by benface** [comment]
A way to quickly disable/enable all plugins would be great for debugging purposes!

**Comment by engram-design** [comment]
To be fair, you can use `disabledPlugins => '*'` in your `general.php` for debugging. But with many Craft 2 > 3 upgrades, there's either brand-new plugins, or plugins that have changed maintainer that need to be enabled one-by-one.
